src/main/java/Duke.java

```diff
@@ -1,10 +1,261 @@
+            }
+        }
+
+    }
+
+    public static void printTaskList() {
+        System.out.println(line + "Here are the tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            int taskIndex = i + 1;
+            System.out.println(taskIndex + ". " + tasks.get(i).printTask());
+        }
+        System.out.println(line);
+    }
+
+    public static void markTask(String input) throws NumberFormatException {
+        int taskIndex = Integer.parseInt(input.substring(4).trim());
+        if (taskIndex >= 1 && taskIndex <= tasks.size()) {
+            tasks.get(taskIndex - 1).markAsDone();
+            System.out.println(line + "Task " + taskIndex + " marked as done:\n" +
+                    tasks.get(taskIndex - 1).printTask() + '\n' + line);
+        } else {
+            System.out.println("Task " + taskIndex + " not found. Please try again.\n" + line);
+        }
+    }
+
+    public static void unmarkTask(String input) throws NumberFormatException {
+        int taskIndex = Integer.parseInt(input.substring(6).trim());
+        if (taskIndex >= 1 && taskIndex <= tasks.size()) {
+            tasks.get(taskIndex - 1).markAsUndone();
+            System.out.println(line + "Task " + taskIndex + " marked as not done yet:\n" +
+                    tasks.get(taskIndex - 1).printTask() + '\n' + line);
+        } else {
+            System.out.println("Task " + taskIndex + " not found. Please try again.\n" + line);
+        }
+    }
+
+    public static void createTodo(String input) throws IndexOutOfBoundsException {
+        ToDo todoTask = new ToDo(input.substring(5));
+        tasks.add(todoTask);
+        System.out.println(line + "Great! I've added this task:\n" + "   " + todoTask.printTask());
+        System.out.println("Now you have " + tasks.size() + " task(s) in the list.\n" + line);
+    }
+
+    public static void createDeadline(String input) throws IndexOutOfBoundsException {
+        if (!input.contains("/by") || input.indexOf("/by") + 4 > input.length()) {
+            throw new StringIndexOutOfBoundsException();
+        }
+        String deadlineDate = input.substring(input.indexOf("/by") + 4);
+        Deadline deadlineTask = new Deadline(input.substring(9, input.indexOf("/by")), deadlineDate);
+        tasks.add(deadlineTask);
+        System.out.println(line + "Great! I've added this task:\n" + "   " + deadlineTask.printTask());
+        System.out.println("Now you have " + tasks.size() + " task(s) in the list.\n" + line);
+    }
+
+    public static void createEvent(String input) throws IndexOutOfBoundsException, EventTimingException {
+        if (!input.contains("/from") || !input.contains("/to") || input.indexOf("/from") + 6 > input.length()
+                || input.indexOf("/to") + 4 > input.length()) {
+            throw new StringIndexOutOfBoundsException();
+        } else if (input.indexOf("/from") > input.indexOf("/to")) {
+            throw new EventTimingException();
```

Comment:
I noticed you make several references to your command specifiers such as `/from` and `/to`. You can perhaps consider assigning these values to a variable then referencing them in order to minimise the use of magic literals.